webapp/src/client/client.js

```diff
@@ -15,15 +17,26 @@ export default class Client {
     };
 
     doPost = async (url, body, headers = {}) => {
-        headers['X-Requested-With'] = 'XMLHttpRequest';
+        const options = {
+            method: 'post',
+            body: JSON.stringify(body),
+            headers,
+        };
 
-        const response = await request.
-            post(url).
-            send(body).
-            set(headers).
-            type('application/json').
-            accept('application/json');
+        options.headers['X-Requested-With'] = 'XMLHttpRequest';
```

Comment:
This line can be removed, because `getOptions` already adds this header https://github.com/mattermost/mattermost-redux/blob/3d1028034d7677adfda58e91b9a5dcaf1bc0ff99/src/client/client4.ts#L444